### DISCUSSION TITLE
issue is configure openssl 1.1.1v for windows

### DISCUSSION THREAD

**Discussion by Ganeshkumhar1** [openssl/openssl#21773] [Q&A]
I am trying to build openssl 1.1.1v on windows and I am hitting when running the Configure -

>perl.exe Configure  no-asm VC-WIN64A
Configuring OpenSSL version 1.1.1t (0x1010114fL) for VC-WIN64A
Using os-specific seed configuration
Creating configdata.pm
Creating makefile
Possible unintended interpolation of @buildfile in string at configdata.pm line 15452.
Possible unintended interpolation of @buildfile in string at configdata.pm line 15452.
Possible unintended interpolation of @buildfile in string at configdata.pm line 15452.
Possible unintended interpolation of @buildfile in string at configdata.pm line 15452.
Possible unintended interpolation of @buildfile in string at configdata.pm line 15452.
Possible unintended interpolation of @buildfile in string at configdata.pm line 15452.
Possible unintended interpolation of @tmpl in string at configdata.pm line 15452.
Possible unintended interpolation of @tmpl in string at configdata.pm line 15452.
Possible unintended interpolation of @tmpl in string at configdata.pm line 15452.
Bareword found where operator expected at configdata.pm line 15525, near "option, and you will get the information you ask for.  See L</OPTIONS"
  (Might be a runaway multi-line // string starting on line 15452)
Global symbol "@buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "@buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "$buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "@buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "$buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "@buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "@buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "@buildfile" requires explicit package name at configdata.pm line 15452.
Global symbol "@tmpl" requires explicit package name at configdata.pm line 15452.
Global symbol "@tmpl" requires explicit package name at configdata.pm line 15452.
Global symbol "@tmpl" requires explicit package name at configdata.pm line 15452.
configdata.pm has too many errors.
Compilation failed in require.
BEGIN failed--compilation aborted.

It seems generated configdata.pm has syntax errors. could someone please help on this.

**Comment by tom-cosgrove-arm** [comment]
Hmm, I am unable to reproduce this.

I downloaded `https://www.openssl.org/source/openssl-1.1.1v.tar.gz` into a temporary directory, unpacked it, then ran your configure line

```
C:\Users\Tom\tmp>tar xfz openssl-1.1.1v.tar.gz

C:\Users\Tom\tmp>cd openssl-1.1.1v

C:\Users\Tom\tmp\openssl-1.1.1v>dir
 Volume in drive C is SYSTEM                        
 Volume Serial Number is 940A-7559                  
                                                    
 Directory of C:\Users\Tom\tmp\openssl-1.1.1v       
                                                    
01/08/2023  14:51    <DIR>          .               
01/08/2023  14:51    <DIR>          ..              
01/08/2023  14:51                87 ACKNOWLEDGEMENTS
01/08/2023  14:51    <DIR>          apps            
01/08/2023  14:51             2,549 appveyor.yml
01/08/2023  14:51               854 AUTHORS
01/08/2023  14:51             4,099 build.info
01/08/2023  14:51           613,733 CHANGES
01/08/2023  14:51            28,664 config
01/08/2023  14:51             2,510 config.com
01/08/2023  14:51    <DIR>          Configurations
01/08/2023  14:51           130,510 Configure
01/08/2023  14:51             3,627 CONTRIBUTING
01/08/2023  14:51    <DIR>          crypto
01/08/2023  14:51    <DIR>          demos
01/08/2023  14:51    <DIR>          doc
01/08/2023  14:51    <DIR>          engines
01/08/2023  14:51    <DIR>          external
01/08/2023  14:51            11,055 e_os.h
01/08/2023  14:51                84 FAQ
01/08/2023  14:51    <DIR>          fuzz
01/08/2023  14:51    <DIR>          include
01/08/2023  14:51            57,824 INSTALL
01/08/2023  14:51             6,121 LICENSE
01/08/2023  14:51    <DIR>          ms
01/08/2023  14:51            46,758 NEWS
01/08/2023  14:51             4,492 NOTES.ANDROID
01/08/2023  14:51             2,093 NOTES.DJGPP
01/08/2023  14:51             4,578 NOTES.PERL
01/08/2023  14:51             5,532 NOTES.UNIX
01/08/2023  14:51             3,861 NOTES.VMS
01/08/2023  14:51             7,488 NOTES.WIN
01/08/2023  14:51    <DIR>          os-dep
01/08/2023  14:51             3,157 README
01/08/2023  14:51            16,069 README.ENGINE
01/08/2023  14:51                61 README.FIPS
01/08/2023  14:51    <DIR>          ssl
01/08/2023  14:51    <DIR>          test
01/08/2023  14:51    <DIR>          tools
01/08/2023  14:51    <DIR>          util
01/08/2023  14:51    <DIR>          VMS
01/08/2023  14:51    <DIR>          wycheproof
              23 File(s)        955,806 bytes
              19 Dir(s)  540,334,137,344 bytes free

C:\Users\Tom\tmp\openssl-1.1.1v>perl Configure no-asm VC-WIN64A
Configuring OpenSSL version 1.1.1v (0x1010116fL) for VC-WIN64A
Using os-specific seed configuration
Creating configdata.pm
Creating makefile

**********************************************************************
***                                                                ***
***   OpenSSL has been successfully configured                     ***
***                                                                ***
***   If you encounter a problem while building, please open an    ***
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL file first)         ***
***                                                                ***
**********************************************************************
```

The Perl errors you are seeing show signs that a stray open quote or other string start sequence is present higher up in the file - maybe a backslash or something has got in from a previous attempt to configure?

I suspect the cause of the problem is something typed before the configure line you've shown us (because that does actually work).

Can you try again in a clean directory, running just the line you have given?

**Comment by t8m** [comment]
What Perl are you using? Currently the recommended Perl for Windows that we use for testing the Windows support is the Strawberry Perl.

**Comment by Ganeshkumhar1** [comment]
Thanks Tom, Tomas,
I was able to figure out that it was 2 different Perl in set in env causing the issue. I used Strawberry Perl and it worked fine.
